### PR TITLE
Pandas 2.2; remove some deprecated behavior; update black+pyright

### PR DESCRIPTION
**Pull request by twoertwein**
Also reduced the number of failures on nightly, but it still fails.

I should have put the black changes in a different commit, but I realized that too late.

edit: run black on a new branch, rebased, and force pushed to separate black and the other changes

### CODE REVIEW

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -139,39 +139,45 @@ class _iLocIndexerFrame(_iLocIndexer):
...
-        idx: IndexType
-        | MaskType
-        | tuple[IndexType | MaskType, IndexType | MaskType]
-        | tuple[slice],
+        idx: (
+            IndexType
+            | MaskType
+            | tuple[IndexType | MaskType, IndexType | MaskType]
+            | tuple[slice]
+        ),
```
Comment (twoertwein) [inline pandas-stubs/core/frame.pyi:147]: (Not too keen on the new formatting)

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein . 

I had to resolve some merge conflicts.

**Comment by twoertwein** [comment]
Thank you for taking care of the conflicts!

Can probably have a new release - or wait for the open PRs!

**Comment by Dr-Irv** [comment]
> 
> Can probably have a new release - or wait for the open PRs!

I'm going to wait for the open PR's.

Will you see if you can get nightly working?

**Comment by twoertwein** [comment]
I think I can get that done this evening. How do we want to handle breaking changes in 3.0 (that do not warn on 2.2 but fail on 3.0)? Ignore in the CI or proactively remove them from the stubs? I'm fine with either.

**Comment by Dr-Irv** [comment]
> I think I can get that done this evening. How do we want to handle breaking changes in 3.0 (that do not warn on 2.2 but fail on 3.0)? Ignore in the CI or proactively remove them from the stubs? I'm fine with either.

Don't change the stubs for nightly.  Just change the `test` modules so that they avoid/change the test if the version is larger than 2.2.99.

In `tests/__init__.py`, we have this:
```python
PD_LTE_21 = Version(pd.__version__) < Version("2.1.999")
```
So you'd change that to now be `PD_LTE_22` using version `2.2.999` and then if some test is inappropriate for nightly, just surround it with `if PD_LTE_22` so that it runs with 2.2., but is skipped on nightly.

**Pull request merged**